Introduction to the course
==========================

Who are we? Who am I?
---------------------
- Different academic backgrounds (IBL, CML)
- Different interests, needs, and expectations
- Different skill levels and experiences

Prerequisites and assumptions
-----------------------------
- We are biologists: we are aware of the principles and 
  practices of molecular biology, biogeography, systematics,
  and evolution
- We are natural scientists: we've had previous training
  in statistics (e.g. what is a PCA? What is Bayesian 
  statistics?), know how to read scientific publications, 
  and know how to present research, orally and in writing
- We are not computer scientists, but we're not afraid of
  computers
- We are going to learn together: questions, responses,
  discussions, interruptions, are always welcome

Learning goals
--------------
- To develop a data-centric view of biodiversity research
- To adopt principles and practices of open science
- To learn computational skills in biodiversity analysis
- To improve our communication abilities

Course outline
--------------
Lectures in the morning, practicals in the afternoon, 
presentation, report, exam.

### [Week 1: Molecular Biodiversity](../)
- **Lecture topics**: DNA sequencing techniques; Barcoding; 
  Metabarcoding; Phylogenetics
- **Homework**: _Open Science, Open Data, Open Source_
- **Practical**: Analysis of mycorrhizal molecular diversity
- **Presentations**: 10 minute standup about a topical paper

### [Week 2: Geospatial Patterns](../../week2)
- **Lecture topics**: GIS and the geographical approach;
  Data input, management, and analysis; Niche modeling
- **Homework**: Collecting occurrence data from GBIF
- **Practicals**: ArcGIS, MAXENT
- **Report**: Niche modeling results for GBIF species

### [Week 3: Comparative Analysis](../../week3)
- **Lecture topics**: Trait diversity; Tree topologies;
  Comparative character analysis; Diversification
- **Practicals**: RMarkdown, Phylogenies in R (tree shape,
  diversification), Likelihood and Bayesian ancestor 
  reconstruction, trait analysis

### [Week 4: Wrap-up and Exam](../../week4)
- **Lecture topics**: Diversity in space, time, and function
- **Exam**

Links to other courses
----------------------

- **Genomic Architecture**: the substance of GA is assumed 
  to be a given. We will give more attention to the data
  science and bioinformatics side of genome (and other)
  sequencing.
- **Advanced Statistics**: also assumed to be a given. We
  will refresh our R skills and apply them more specifically
  to cases involving biodiversity.
- **(old) Methods In Biodiversity Analysis**: is more or less
  renamed as Integrative Taxonomy. "Old MIBA" is therefore a
  different course, with different contents.
- **Integrative Taxonomy**: is mostly "Old MIBA" under a new
  name. Focuses more on taxonomy and systematics, less on 
  data science and informatics.

Teaching materials
------------------

- Slides (re-formatted as handouts) are at http://github.com/naturalis/mebioda
- Handouts and slides are subject to ongoing (wiki-like) improvement
- During the course we will learn how to use this platform to
  share data, scripts, files with each other

Teamwork
--------

- With the hands-on parts of the course we will try to learn in pairs,
  helping each other
- All assignments are individual, though

Grading
-------
- Exam: 50%
- Paper presentation week 1: 20%
- Report week 2: 20%
- Participation: 10%